### ISSUE TITLE
Android App V3 constantly requiring login details

### ISSUE THREAD

**Issue by buggerlugs** [nextcloud/android#2133]
### Actual behaviour
Asks for login details, you enter them.  Opens the app but after a few seconds it pops up a screen telling you to Grant Access to your device.  Tap that and you get the login screen.  You enter your login details.  The process repeats continuously.

### Expected behaviour
To login once and use
 
### Steps to reproduce
1. Enter user name
2. Enter Password
3. Wait a few seconds

### Environment data
Android version:  5.02 and 5.1.1and 7

Device model: 
Samsung SM-T365, SM T-555, SM -T535 all from Samsung

Stock or customized system: Stock

Nextcloud app version: 3

Nextcloud server version: 12.0.3

**Comment by wschmidhuber** [comment]
Same here on SM-T825 after update to Nextcloud Android App V3
Server version Nextcloud 12.0.3 - behaviour didn't change after update to 12.0.4
Devices with previous version of Android App still doing fine.

**Comment by RayOfLight1** [comment]
Confirmed blocking bug on Samsung Galaxy Note 3 (SM-N9005), what a way to get an update, I needed to access some files miles away from home and had this bug happen to me, asked myself what went wrong, opened Google Play, saw Nextcloud as latest updated, acknowledged the source of the problem.

**Comment by rolandixor** [comment]
Confirmed - Blue Vivo 5R running Android 7.0
It keeps redirecting me to the old login method.

**Comment by matakapo** [comment]
Confirmed - one plus 3T running Android 8.0

**Comment by danielb42** [comment]
Confirming for Android 7.1.1 on Motorola Z Play (XT1635-02), App v3.0.0, Server v12.0.3.

**Comment by acekrystal** [comment]
Confirmed - Samsung Galaxy S8 here.

Bit different though. after the update it contantly gives me the login screen where I only need to fill in the password. After good password submit I get to the usual main folder view, but then after 1,5 second the loging screen comes up again, asking me again for my password. It does see I fill in the correct password, as when I type it wronge it will actually tell me my password is wrong.

Login on windows client, and browser do work normally with same account.

So it is a login loop (or it try's to get authentication for every item in your nextcloud?)
On the PlayStore you see tons of people reporting this bug with a negative review.

This should be fixed ASAP!

**Comment by mr-bolle** [comment]
I use android app v3 RC3 and can successfully login into the server v12.0.5 (with the authentication token) 

With the server v13 final, i can't login with the authentication token, the application uses the old authentication method.

**Comment by georgep56** [comment]
Confirmed - same error with Xiaomi Redmi Note 3 running MIUI 9.2 connecting to Nextcloud server v10.0.1 plugin on FreeNAS 11.1.

**Comment by mr-bolle** [comment]
@georgep56 could you update your server. Nextcloud 10 are out off support.

**Comment by georgep56** [comment]
Unfortunately the plugins for FreeNAS 11 are very out of date :-/
I installed the plugin to evaluate nextcloud and I'm building v13 in a
separate jail right now :-)

**Comment by dxj728** [comment]
Confirm that app: 3.0.0 android 7.0 server: 13.0.0 has repeated logon and app is almost unavailable, I hope someone can solve this problem.

**Comment by skyfox675** [comment]
Confirmed on Pixel 2 Xl, v3.0.0.

Thanks, 
-Skyfox

**Comment by EazyDefcon** [comment]
Same problem here. Nexus 6 Android 7.1. Server version 12.

**Comment by georgep56** [comment]
I just built a new server instance using 13.0.0 and everything appears to
work OK.

**Comment by fortytwome** [comment]
I had this bug with a 12.0.2 Nextcloud server. Upgrading the server to 13 solved the Android bug.

**Comment by Jas2Ma** [comment]
Same here with Server 12.0.2 and FP2, Android 6.0.1

**Comment by user23498723452** [comment]
This is impacting my Google Pixel on Android 8.1.  Server version 12.0.0

**Comment by ehrenbe4** [comment]
Same Problem in Galaxy S3 i9300 with Server 12.0.0

**Comment by cpranzl** [comment]
In respect of my closed issue #2147 I want to add:

Confirmed on Motorola G5 with Android 7.0 and Nextcloud-App 3.0 (as well as Passman, Nextcloud/ownCloud News Reader and DAVdroid).

I started with an 12.0.5 and updated to 13.0.0. this morning in hope to relsolve this problem like @georgep56

**Comment by AndyScherzinger** [comment]
@cpranzl thanks for the details. Looking at the fact that all your Android apps are failing to login I'd guess it is a server issue as in problems with urls/routes (apache/nginx config).

**Comment by cpranzl** [comment]
@AndyScherzinger thats what i thought at first, but strangely the Nextcloud Desktop Sync Client on my Laptop is working flawless and i'm also able to connect via browser (Google Chrome 64.0.3282.140, Microsoft Edge 40.15063.674.0, Microsoft IE 11.850.15063.0).

Another clue that the problem lies elsewhere is that new Nextcloud-App (3.0) was able to download my login image and displayed it.

**Comment by AndyScherzinger** [comment]
cc @MorrisJobke any ideas?

@cpranzl the windows client might use other urls (don't know that actually) and the browser definitely uses other URLs or might get redirected since the Android client communications via json most of the time.

**Comment by wschmidhuber** [comment]
@AndyScherzinger - Not only Desktop Clients and Browsers work fine but also Android devices with clients not yet updated to 3.0 don't show any problems here. It all started with 3.0 and there are plenty of similar complaints in Google Playstore Comments on Nextcloud.

**Comment by cpranzl** [comment]
Thats what I see in my nextcloud.log:

`{"reqId":"3IQ5EdwLIfjsGcmkNrqa","level":2,"time":"2018-02-12T12:01:58+00:00","remoteAddr":"xxx.xxx.xxx.xxx","user":"--","app":"core","method":"HEAD","url":"\/remote.php\/webdav\/Photos\/uploads\/2018\/02\/","message":"Login failed: 'yyyyyy' (Remote IP: 'xxx.xxx.xxx.xxx')","userAgent":"Mozilla\/5.0 (Android) Nextcloud-android\/3.0.0","version":"13.0.0.14"}`

**Comment by mario** [comment]
I think this is now fixed. Sorry folks, will be in 3.0.1 - cc @tobiasKaminsky @AndyScherzinger.

https://github.com/nextcloud/android-library/pull/120

**Comment by mario** [comment]
(At least I hope this is it :P)

**Comment by tobiasKaminsky** [comment]
To make really sure this works prior releasing 3.0.1, can some of you create us a test account and send the credentials to android at nextcloud dot com with a reference to this issue?
Please reply that you send it, I guess 2-3 are enough.

**Comment by Jas2Ma** [comment]
I send you credentials.

**Comment by tobiasKaminsky** [comment]
@Jas2Ma thank you for your credentials.
On a fresh device I installed 2.0.1, logged in, upgraded to 3.0 and there was no login screen.
It is working just fine.

@mario do you have an idea? (as I used stable-3.0.0 tag, I did not have the fix included, but still it works)

**Comment by tobiasKaminsky** [comment]
@all what kind of devices and which android version do you use?

**Comment by RayOfLight1** [comment]
nextcloud 12.0.3, galaxy note 3 with lineageOS 14.1 (7.1.1/7.1.2), using
nextcloud app 3.0.0

Atentament:
Jordi Paradell
DEEJAY.JPM@gmail.com

2018-02-12 15:43 GMT+01:00 Tobias Kaminsky <notifications@github.com>:

**Comment by danielb42** [comment]
@tobiasKaminsky 
> On a fresh device I installed 2.0.1, logged in, upgraded to 3.0 and there was no login screen.
It is working just fine.

The problem does not occur until the device is rebooted. (Sorry for not mentioning this detail sooner.)

**Comment by Jas2Ma** [comment]
@tobiasKaminsky

yesterday we uninstalled the App and returned to 2.0.1. Because of your remark I upgraded again and have no problems atm.

@danielb42: Was the same for me, be after reinstalling everything works fine even after reinstalling.

**Comment by tfbiii** [comment]
Confirmed - Pixel 2xl running 8.1.0 NextCloud Android v3

**Comment by wschmidhuber** [comment]
@tobiasKaminsky 

Server: 12.0.4

1) Galaxy Tab S3 SM-T825, Android 7.0, login-loop since update to App 3.0, also when re-installed with default settings, 

2) Galaxy S7 SM-G930F, Android 7.0, using external SD memory, still working fine with App 2.0.0

**Comment by user23498723452** [comment]
my storage path says /storage/emulated/0

I'm using default storage config for my Google pixel on android 8.1. nc server 12.0.0

**Comment by AndyScherzinger** [comment]
Login issues reported via the forum with Nc11 and App v3:

Unfortunately you updated the app to version 3.0.0 and shot all connections to (at least) nextcloud 11. According to logfile the login works:
```
{"reqId":"SUml+GU6JIY2QoKjugq3","remoteAddr":"188.193.244.YY","app":"admin_audit","message":"Login successful: \"frank@ithnet.com\"","level":1,"time":"2018-02-11T14:02:38+00:00","method":"POST","url":"\/ocs\/v2.php\/apps\/notifications\/api\/v2\/push?format=json&pushTokenHash=9646568dc2c20505b98b317bf3956c76a4d632c1eaa52b4ace3c0959fa40a0758cc6663f6137d39eae090bf8a72ef4e2d5fedf8e8e181092bf3410c78d461c14&devicePublicKey=-----BEGIN%20PUBLIC%20KEY-----%0AMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAiO8R5ovEN6zHs8%2F5phEr%0AwFj%2B03Gq%2BfKR6Yq4V3qNdqxuc%2F5yBClsejTJOaoCxAUbkM5VpitznRSFjwQESHvP%0ArxL3OISZepRBAJZLDq3FWV24WGLcXQFZPoN3m%2BqYIScQK6Q4IYGB%2F3bIvABhiav4%0A2fffGcSyywuiROPn4pjECZd6PPQ8PAqE%2Fl5jZpvibM9JuPGcbe%2F%2Bm1H8TVXAWTUh%0ALfOt8BKI2LdJeaBU9k4MmDtZZ3JCk78nYgV6C3Xw4mSM58C9b8aX6I1xPCJErbsK%0AipA7OG7ixCMAB%2F2pN1FGQzSEC1PSyEb3RSRFKT8VfJJWmBSkzjTFInDR0tA0HKwC%0AGwIDAQAB%0A-----END%20PUBLIC%20KEY-----%0A&proxyServer=https%3A%2F%2Fpush-notifications.nextcloud.com","user":"frank@ithnet.com","version":"11.0.5.1"}
```
But then:
```
{"reqId":"sghjUlD\/8F\/aFr1yqZA6","remoteAddr":"188.193.244.YY","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 401 CSRF check not passed.\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotAuthenticated\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/user-local\\\/cloud.default\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Auth.php(154): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Auth->auth(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/srv\\\/user-local\\\/cloud.default\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php(199): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Auth->check(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#2 \\\/srv\\\/user-local\\\/cloud.default\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php(150): Sabre\\\\DAV\\\\Auth\\\\Plugin->check(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#3 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/srv\\\/user-local\\\/cloud.default\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#5 \\\/srv\\\/user-local\\\/cloud.default\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(466): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#6 \\\/srv\\\/user-local\\\/cloud.default\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#7 \\\/srv\\\/user-local\\\/cloud.default\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(61): Sabre\\\\DAV\\\\Server->exec()\\n#8 \\\/srv\\\/user-local\\\/cloud.default\\\/owncloud\\\/remote.php(165): require_once('\\\/srv\\\/user-local...')\\n#9 {main}\",\"File\":\"\\\/srv\\\/user-local\\\/cloud.default\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Auth.php\",\"Line\":220,\"User\":\"frank@ithnet.com\"}","level":0,"time":"2018-02-11T14:02:57+00:00","method":"PROPFIND","url":"\/remote.php\/webdav\/","user":"frank@ithnet.com","version":"11.0.5.1"}
```
All from the same IP with the same account, 20 seconds later failed.
Why is that?

**Comment by etaijoverlap** [comment]
Confirmed on Samsung Galaxy S8, Nextcloud 12. Same pattern.

**Comment by dxj728** [comment]
I seem to find some clues that whenever I log in to a connection with a domain name, it keeps flashing back in, but when I log in using IP, it seems to be all right

**Comment by tobiasKaminsky** [comment]
I suspect that you the cause of this is
`
01-09 14:33:02.028 1155 2923 W AccountManagerService: insertAccountIntoDatabase: Account {name=user@cloud.server.com, type=nextcloud}, skipping since the account already exists
`
Logging in with IP address instead of server url will create a new account.
The main questions still remains why is login popping up at all: why does android app think you need to re-authenticate.

Can you provide us additional infos via logcat? Maybe we see there a reason for this
https://github.com/nextcloud/android/blob/master/README.md#getting-debug-info-via-logcat

**Comment by tobiasKaminsky** [comment]
Can some of you try to update to latest stable 12.0.5?

**Comment by tobiasKaminsky** [comment]
Background of this is, that due to a bug, we send a "wrong"/too new user agent, which will lead the server to refuse it.
This is fixed with 12.0.5 release, which allows both user agents.

In parallel android client 3.0.1 will restore the old behaviour, but to narrow down the issue and to verify that the user agent is the only problem, we need some users to upgrade 12.0.5.

**Comment by Micha-Btz** [comment]
I have updated android app from 2.0.1 to 3.0.0 with nc 12.0.5. No problems I don't have to reauthenticity the login.

**Comment by bastei** [comment]
@tobiasKaminsky This could be somehow related to #1531: The Android PackageHelper [doesn't respect](https://android.googlesource.com/platform/frameworks/base.git/+/android-8.1.0_r9/core/java/com/android/internal/content/PackageHelper.java#486) `android:installLocation` during package upgrades. Thus, after an update even v3.0.0 can reside in the external storage with the complications described in #1531.

At least the following [comment](#issuecomment-364945529) sounds like this problem:
> The problem does not occur until the device is rebooted.

To determine if that is the problem for some of the users here, it would be helpful to know if they see the Nextcloud app in Android Settings -> Accounts.

**Comment by wschmidhuber** [comment]
On my Galaxy Tab S3 SM-T825, Android 7.0, where the login-loop occurs since update to App 3.0 the App is shown in Accounts (but can't synchronize). I uninstalled and reinstalled App 3.0 so all settings should be default. And I didn't reboot the device before the login-loop occurred first.

**Comment by RayOfLight1** [comment]
I updated nextcloud from 12.0.3 to 13.0.0 and as of now, everything is
working back again with nextcloud-android 3.0.0

2018-02-13 12:46 GMT+01:00 wschmidhuber <notifications@github.com>:

**Comment by tobiasKaminsky** [comment]
@bastei which device do you use? 
To prevent this, we even introduced https://github.com/nextcloud/android/blob/stable-3.0.0/src/main/AndroidManifest.xml#L76 which should in fact be not necessary as the default is to have it installed only internal.

**Comment by haverholm** [comment]
The login loop on the Android client prompted me to update my server instance to v13. Instead of a *constant* loop I now only have to reauthenticate every time a periodic sync fails. 

I'm on a Oneplus One running Lineage OS 14.1, Android 7.1.2

If it helps, see attached screendump of the log. 
![nextcloud_log](https://user-images.githubusercontent.com/5727354/36149778-58380702-10c1-11e8-8009-fb4131fb69ca.jpg)

Hoping v3.0.1 will be pushed to F-droid as fast as possible!

**Comment by wschmidhuber** [comment]
Solved here. For my Galaxy Tab S3 SM-T825, Android 7.0, where the login-loop occured since update to App 3.0, updating from Server 12.0.4 to 12.0.5 solved the problem.

**Comment by tobiasKaminsky** [comment]
@haverholm Can you create us a test account and send the credentials to android at nextcloud dot com with a reference to this issue?

@wschmidhuber thank you for confirming that upgrading to 12.0.5 fixes it.

**Comment by haverholm** [comment]
@tobiasKaminsky Done, check your inbox.

**Comment by DAveShillito** [comment]
I have the same problem
Android client 3.0.0
Wileyfox Spark+
Android 7.0

I'm pretty sure the app is not on external storage on the phone.
Settings->Apps->Nextcloud says 505MB used in internal storage, of which App is 26MB and data 479MB.
Likewise the storage path is /storage/7369-B726/Android/data/com.nextcloud.client/files which I think means it is on internal storage on the phone.

I'm connecting to Nextcloud 11.0.3 running on a Raspberry Pi2.
I tried updating this to 12.0.5 but after the update I was not able to access Nextcloud from the web interface at all so I had to revert back to the pre-upgrade backup.

This is only my home server, so not mission critical and I will try to update the server when I can, but it would be nice to get the android app working, or temporarily disable it trying to auto-upload photos and backups. Calender and contact syncing using WebDav seems unaffected.

**Comment by user23498723452** [comment]
@tobiasKaminsky I do see nextcloud in the list of accounts on my Google pixel.

**Comment by user23498723452** [comment]
@tobiasKaminsky Found a possible fix!  my nextcloud is hosted at home and I saw the issue when trying to access the site using a public dns name. which works fine from a routing perspective.

I've just tested for the first time from the internet, and I was prompted to enter my username/password as needed.

it appears to have worked.  I'll update if login prompts continue.

**Comment by tobiasKaminsky** [comment]
@DAveShillito a "random" storage name is often the external sdcard, whereas the internal storage is referenced as /storage/emulated/(0)
As mentioned, we will provide 3.0.1 soon. Or 12.0.5 will fix it also. Maybe it is worth to investigate the problem of 12.0.5.

@user23498723452 I think this is another problem. I discussed it earlier this day on IRC. The web browser on android could also access the NC website, but not android client.
As this is not reproducable in every scenario, I could not provide a fix :-/

**Comment by DAveShillito** [comment]
@tobiasKaminsky Thanks for the quick reply.
The "random" storage name initially made me think it was SD storage, but I was confused that Settings->Apps->Nextcloud says 505MB used in internal storage.
Not that it matters, I only mentioned it in case it helped your diagnosis.

As I said I'm not mission critical so I can wait for 3.0.1.
I will definitely be looking at getting 12.0.5 working (or even jumping to 13) but that looks to be a bigger issue for me to fix with a manual update, or starting my server from scratch again. I just tried to do it quickly using the updater app to see if it fixed the login issue.

Thanks to everyone working on this.
Generally my experience with Nextcloud and the Android client has been flawless over the many years I've been using it, unfortunately these minor hiccups occur, but you are all obviously on them.  
Thanks again. :)

**Comment by bastei** [comment]
> @bastei which device do you use?

I use an (unofficial) LineageOS 14.1 installation. I left a [note](1531#issuecomment-365330218) at the other ticket concerning the app location after updates. Seems that this is not the main culprit of the problems discussed here, but I wanted to make you aware that this could lead to similar problems.

**Comment by user23498723452** [comment]
@tobiasKaminsky not sure what you mean. I'm testing this issue from home again and it has returned.  I did reboot the phone to be sure.  it seemed to be working before I rebooted it.

**Comment by tobiasKaminsky** [comment]
@haverholm I could not reproduce it. I could sync, download/upload files and even left the device on for ~2 hours, but never got a log out :-/

**Comment by haverholm** [comment]
@tobiasKaminsky And in the meantime the loop has ceased here, too. I did a LineageOS system update, but surely that isn't what fixed it? smh

**Comment by tobiasKaminsky** [comment]
I only had this one time during testing and there the error message in logcat was:
> AccountManagerService: insertAccountIntoDatabase: Account {name=user@cloud.server.com, type=nextcloud}, skipping since the account already exists

So maybe this was resolved / cleaned via the system upgrade.

**Comment by tobiasKaminsky** [comment]
3.0.1 is now released.
Please test and report back if it is working for you

**Comment by DAveShillito** [comment]
@tobiasKaminsky I just downloaded the 3.0.1 update and all seems good.
My previously stalled auto uploads have completed, and I can browse files without the login screen popping up.
Thanks to everyone for the quick fix :)

**Comment by mr-bolle** [comment]
@tobiasKaminsky where I could download the latest version? On f-droid and Google play I find only the 3.0.0

**Comment by Siggi0904** [comment]
with Lineageos 7.1.2 on NC 11.0.1 and 3.0.1 android app it won't work.
In login window "Secured connection established" is shown. Then I input the password and after connect button Message "Couldn't found host" is shown. Both here in german, means translation could not exact the same.

**Comment by AndyScherzinger** [comment]
@mr-bolle you should see it on GPlay while f-droid will likely need some days to build and publish.

**Comment by tobiasKaminsky** [comment]
@Siggi0904 this seems to be another problem.
Please open up a new issue.

**Comment by xXSTrikeXx** [comment]
Since @DAveShillito tested the new version successfully and @Siggi0904 has got -I think- another error (I guess dns settings, so please open a new one), we can close the issue. Just feel free to reopen if someone got some errors.

**Comment by acekrystal** [comment]
I'm still on app version 3.0.0 but it started working for me just now while getting some other problems fixed. Nextcloud version 12.0.5.

The 2 things I did different was : 
1. I cleaned my Database Table "oc_bruteforce_attempts" as X-Forward seems to not work yet and it caused my login to be really slow (40sec+). So It could actually have been a time out on the app side.
2. I got a login screen on the nextcloud app where I had a little check box to "stay signed in". I used that for a change, and for now it suddenly started working.

Hope this might give some directions.
Will try the app 3.0.1 soon too, but since I saw I was still on 3.0.0 I thought it might be worth sharing.

**Comment by user23498723452** [comment]
3.0.1 appears to have fixed the issue.  I've given it a couple of days to be sure.

**Comment by AndyScherzinger** [comment]
3.0.1 fixes one issue that led to the recurring logins which only happens in that scenario when the server version is < 12.0.5. So 3.0.0 won't have the issue in case the server is up-to-date.

**Comment by user23498723452** [comment]
To be clear,  this was the most frustrating bug as it preventing me from accessing or syncing my files. 

Not only could I not use the app, but I couldn't fix the issue as there is no way to roll back on a Google Play store apk.  Sure, one could install an untrusted apk, but that isn't acceptable for many.  I'm unaware of any supported copies of all the actual APKs released through the play store.

 While I appreciate the efforts of the NC dev team, it seems this points out some failure to robustly test all of your supported versions.  Only testing the latest version is not acceptable and I hope that NC learns the importance of this.

Personally,  I'm putting this app back on to the list of apps requiring manual update approval.  I cannot go days without being able to sync my files.  Please consider sharing ways clients can roll back to older versions of the Google Play store apk.

**Comment by mr-bolle** [comment]
@AndyScherzinger now I get the update over f-droid, on GPlay this version isn't available for me (Android 7.1.1 Sony Xperia Z5 Stock ROM)

maybe it would be possible to upload this application on https://download.nextcloud.com/

**Comment by AndyScherzinger** [comment]
@mr-bolle @user23498723452 all GPlay releases are available via direct download @ https://download.nextcloud.com/android/ so this way you can manually downgrade since GPlay doesn't provide such an mechanism itself. These apks are signed so they are the same apks as the ones published on GPlay.

> While I appreciate the efforts of the NC dev team, it seems this points out some failure to robustly test all of your supported versions. Only testing the latest version is not acceptable and I hope that NC learns the importance of this.

We didn't just test the latest version. Nc server Life Cycle is support for the last 2 major versions which would have been at the time of publishing Nc 13.0.0 and Nc12.0.5, so we did test the supported versions, while I agree that this issue shouldn't have made it to production. On the other hand there were over 5730 testers via the beta program who ran this release against their servers without reporting login issues. So that would mean they didn't report it or all of them ran 12.0.5+.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky can you publish the latest 3.0.1 on the download site please :)

**Comment by xXSTrikeXx** [comment]
Im in beta channel, too. I've upgraded few days before publishing stable to nc 13.0.0 and I havent got any time a login loop. So well, in todays time every producer have got their own os of a customization from android.  For example huawei with emui. Then differents android versions and so on. So in my opinion feel free to get in the beta team, to improve beta. Then you could be able to compare open source with closed source, for example dropbox or microsoft.

**Comment by Siggi0904** [comment]
Here these bug still exists.
With 3.0.2 the login windows still occours.

Please reopen these bug and check what happend there.

**Comment by xXSTrikeXx** [comment]
@Siggi0904
Please check your table of bruteforce attempts. Maybe this results the behavior. What happens when you use a different device?

**Comment by Siggi0904** [comment]
Brute force table cleaned. Same behavior.

**Comment by janipewter** [comment]
I have this behaviour too. The 3.0.2 update fixed the "could not find host" error but now I constantly get the login box. Put password in and hit OK, logs in, but as soon as I touch any button it goes back to login box.

ownCloud server v10.0.4
NextCloud Android app 3.0.2
Android version 7.0
Samsung Galaxy S7

I can provide a test account. Please let me know where to send the details

**Comment by janipewter** [comment]
I have migrated to Nextcloud server instead. No idea why I wasn't using it in the first place, but now everything works perfectly.

Thanks

**Comment by DAveShillito** [comment]
After the 3.0.2 upgrade this problem has returned for me as well.

Android client 3.0.2
Wileyfox Spark+
Android 7.0

I'm connecting to Nextcloud 11.0.3 running on a Raspberry Pi2.
I do intend to update to Nextcloud 13 when possible, but when I tried updating to 12.0.5 I was not able to access Nextcloud from the web interface at all so I had to revert back to the pre-upgrade backup.

**Comment by sukosevato** [comment]
I can confirm that I had this issue as well with the 3.0.2 nextcloud client for android (with 12.0.x as server). After upgrading to nextcloud 13.0 serverside the issue is gone.

**Comment by gruhland** [comment]
I also have this issue with a Motorola Moto Z (Android 7.1.1) and Nextcloud app 3.0.3. Server version is 11.02.

**Comment by Siggi0904** [comment]
Hi, problem still exists. No solution possible?
Where can I test a NC 12 or NC 13 server with these app?

**Comment by xXSTrikeXx** [comment]
@gruhland 
Nextcloud Server 11.02 will be EOL:
https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule

I suggest to update with the last update (https://help.nextcloud.com/t/nextcloud-13-0-1-12-0-6-and-11-0-8-available-with-improved-password-protection/28970) to 11.0.8 or directly to 13 😉

**Comment by Siggi0904** [comment]
Admin says, it's much work to upgrade from 11 to 12 or thirteen. :-)
Many manual work to and after upgrade.
Hope he will upgrade next time to 13. I'll be in contact with him.

**Comment by xXSTrikeXx** [comment]
@Siggi0904 Every update will also enhance the security 😉 
It´s like Microsoft XP. You can continue to use it, but you won´t get updates, especially security patches for it, since it´s EOL. So it´s a risk on your own security. When you stay at one version, you need also to stay on one version with other components. In this case the android app. You updated it now and now the server is out of date while using the latest version of android client. Thats dangerous. I think the easiest way is to backup all and update atleast to version 12.x.x. Thats not such a big step and work with the webupdater. He doesn´t need to program (& testing) anything, so it´s not so much work 😁

**Comment by c33s** [comment]
@xXSTrikeXx the problem is that my 11.x installation do not show me the update for 12 on the stable channel

**Comment by AndyScherzinger** [comment]
> the problem is that my 11.x installation do not show me the update for 12 on the stable channel

Any idea @MorrisJobke ?

**Comment by MorrisJobke** [comment]
> Any idea @MorrisJobke ?

We put 11.0.8 to the stable channel last Monday to get maybe more people at least to a stable 11 release, because 12.0.x was served before that since autumn last year. We will soon move this to a stable12 release soon (within 1 or 2 weeks). If you want to update now: use the beta channel as there are no betas out right now and this will just give you 12.0.6 ;)

**Comment by tobiasKaminsky** [comment]
I created a test version, which can be installed as an upgrade to the existing app.
Can someone with at least Android 5.1 (Lollipop) can test it?

Please do this steps:
- verify that with current 3.1 version the problem still occurs
- install test version
- try it again

Link to download test version: https://cloud.nextcloud.com/s/fazWbsg8LoAKCtE
(it is signed by us and will replace 3.1 version with 3.1.1, so it will not interfere with upcoming 3.2. You have to enable "allow installation from unknown sources").

**Comment by mr-bolle** [comment]
I will test this behavior the next day. Possible Android Release 6, 7.1 and 8.0

**Comment by yolanda93** [comment]
Hi, 
I am using the Android Nextcloud 3.1.0, Server v11.0.2 and Android 6 and I have exactly the same problem commented here. @mr-bolle, is it a client bug or do I need to update the server? Thanks

**Comment by mr-bolle** [comment]
Hi, @yolanda93 since some hours i try to find the reason to reproduce the problem. I would suggest to upgrade the Server. It is recommended to use the latest stable release 13.0.2.

To you use docker for your nextcloud Server or you run this directly on your host server?

**Comment by Siggi0904** [comment]
Please don't let the guys upgrade the server. It should be fixed with older versions too.
Many installation are at older versions like 11 or 12 and these version should be work too.

It is an android app authentication problem, not a server side problem.

**Comment by mr-bolle** [comment]
@Siggi0904 please remember that the Server v11 are out of support.  >> [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule)

I try different server, on some the authentication works, on some not. (especially with App Token)

**Comment by AndyScherzinger** [comment]
> Please don't let the guys upgrade the server. It should be fixed with older versions too.
Many installation are at older versions like 11 or 12 and these version should be work too.

> It is an android app authentication problem, not a server side problem.

@Siggi0904 this isn't true, well true in some scenarios what will never work again is anything below 11.0.8 same goes for Nc12 if it is on an old patch level for a simple reason: The Android client uses a new user agent string due to the e2ee feature and will thus be blocked by user agent string on older server so it is a server issue and like @mr-bolle said even Nc11 is out of support so these servers will never work again with the Android client (no matter the reason being it client or server).

It is correct to some extend since the client now also changes its Android requirements so it has to be installed on the internal memory for upgraded Android clients that formerly installed it on the sd card this can lead to issues with the client finding the account again (not found in the database) while after a successful re-login the saving of the account fails in the same database since it is already in there (which is then a client issue, yes).
A way to workaround this issue is to either put a / at the end of the URL or change to upper/lowercase writing of the username. For users with rotted phone another (more complex) way is to remove the account manually from the account database but like I said that is more complex and needs technical know-how.

So the solutions is simply and might well be frustrating. If you lock yourself to a certain server version (which is a security risk!) you also lock yourself to a specific client release, for Android that be v2.

On "clean", fresh phones a fresh installation of the app should definitely work with v3 for Nc 11,12,13 them being on their respective, latest patch level.

**Comment by yolanda93** [comment]
I have upgrade the server to 12.0.7 and with this version seems to work fine.

**Comment by AndyScherzinger** [comment]
Glad to hear that @yolanda93 :heart:

**Comment by mr-bolle** [comment]
Hi, 

now i try several Server (i use Docker Container - Link to docker-compose into Server Column)
Often i have Problems to login with App-Token [Screenshot](https://user-images.githubusercontent.com/22414212/40588561-8d8bec7e-61df-11e8-8f83-a882b44937a4.jpg)

_**Note**_: Now i found out that the User Name with Password Access is **not** Case sensitive, and Login with the App Token is Case Sensitive, That's why i get so often the Error 401 during Login with App Token :neckbeard: 


| Server | App | Passwort Login | App Token Login | Device |
|--|--|--|--|-|
|[12.0.4-apache](https://github.com/nextcloud/docker/tree/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/apache) |3.1.0|ok|ok| Samsung S9 v8 |
|[12.0.4-apache](https://github.com/nextcloud/docker/tree/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/apache) |3.1.1 Alpha2|ok|failed| Samsung S9 v8|
|[13.0.2-apache](https://github.com/nextcloud/docker/tree/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/apache) |3.1.0|failed|failed| Samsung S9 v8 |
|[13.0.2-apache](https://github.com/nextcloud/docker/tree/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/apache) |3.1.1 Alpha2|failed|failed| Samsung S9 v8|
|[12.0.4-apache-base](https://github.com/nextcloud/docker/#base-version---apache) |3.1.0|ok|failed| Samsung S9 v8 |
|[12.0.4-apache-base](https://github.com/nextcloud/docker/#base-version---apache) |3.1.1 Alpha2|ok|ok| Samsung S9 v8|
|[12.0.4-apache-base](https://github.com/nextcloud/docker/#base-version---apache) |3.1.0|ok|ok| Sony Z5 v7.1.1 |
|[13.0.2-apache-base](https://github.com/nextcloud/docker/#base-version---apache) |3.1.1 Alpha2|ok|ok| Sony Z5 v7.1.1|

**Comment by MorrisJobke** [comment]
> Note: Now i found out that the User Name with Password Access is not Case sensitive, and Login with the App Token is Case Sensitive, That's why i get so often the Error 401 during Login with App Token

🙈 @rullzer @blizzz Could you confirm this?

**Comment by rullzer** [comment]
@MorrisJobke it is user_id vs loginname fun. (Which on the main backend are somewhat interconnected).

Anyway as far as I know @tobiasKaminsky has patches that will fetch the user_id on login and use that when required.

**Comment by tobiasKaminsky** [comment]
@mr-bolle thank you for testing, so it seems that the approach does not work.

@rullzer afaik we fetch & use user_id only for dav related stuff, so this is independent from login.
Can you open up a new issue, assign it to me, with reproducible stable? <3

**Comment by MorrisJobke** [comment]
> with reproducible stable? <3

I guess he means "reproducible steps" 😉

**Comment by tobiasKaminsky** [comment]
Now that I have an affected device, I spent several hours in debugging and trying to understand the problem.
As always there are so many different combinations, that there may be several causes, but one of it is a bug that does not entirely remove a custom account when uninstalling the app.
Also the move from internal to external storage (and vice vera) can cause this.
This also affects other apps, e.g. Dropbox.

Some technical background:
Androids own account manager has a stale account in its database. When querying if a given account exists, our app gets none back, assuming that we have to add it.
This then will fail as the account manager cannot add an account twice with the same name.
Also programatically removing the old account will not work.
But the stored account name is case sensitive and this is our chance.

The internal account's name is built from the username and server address, e.g "tobias" and "cloud.server.com" --> tobias@cloud.server.com.
The username is the name that the admin gave the user when creating the account.
The login flow returns always the username, e.g. if you enter your email address as login, we will internally build the account with the returned username.
However it seems that the login flow is returning the username exactly as entered.

So if you have an account tobias@cloud.server.com and this is causing the login loop, you can try to login with "TOBI" or "TOBI " or "ToBi" and all should allow you to login.
Please note that due to androids bug all older account names are burnt and will not work, so if you have done this multiple times, it may be that no suitable combination is left.


TL;DR: It is a bug in Androids Account Manager system, which we cannot solve, but only try to circumvent: login with uppercase username or append a whitespace.

**Comment by MorrisJobke** [comment]
> TL;DR: It is a bug in Androids Account Manager system, which we cannot solve, but only try to circumvent: login with uppercase username or append a whitespace.

How about reporting this upstream? 😉

**Comment by McAviti** [comment]
So I read that there is no manual way to remove the stale accounts from the phone? (I've got Android 7.0).
The issue seems to be the case because changing different upper/lower case variants worked for some time, but I think i used them all (very short account name, unfortunately)

**Comment by AndyScherzinger** [comment]
> So I read that there is no manual way to remove the stale accounts from the phone?

You should be able to see the Nextcloud account within Android's settings -> Accounts and shold be able to remove it/them from there + (not sure!) also clear the data of the Nextcloud app via Android's settings -> Apps

**Comment by McAviti** [comment]
Thanx, Andy!

In the account settings there was none for nextcloud, but I deleted the app's data and then login worked immideately.

~klemens

**Comment by Earl-Brown** [comment]
I had this problem, but was able to fix it by creating the account from the Android account manager.

After having the problem, I opened account manager, then removed the failing account, then created a new account - which loaded the NextCloud login page - and after logging in that way, it worked fine.

**Comment by mathijswesterhof** [comment]
The issue is showing on app version 3.3 android version 8.1 (on one+ 5T) together with server version 13.0. 5 , as well as 13.0.6 and 14.0.0.  no stale account inside the phone. multiple fresh installs. normal login works but gets forgotten, app password shows i should login first, if i remove all sessions it just reverts back to 'connect your account' screen, no warning or info pallette. i also observed the 'revert to old login method' is always there, even when allready logged in with normal login. (all methods tried with normal,camelcased and capital username )I hope someone can help with the info provided,

**Comment by pando85** [comment]
Same here: app version 3.5.1

**Comment by uamoti** [comment]
Same issue with app version 3.11.0 on OP6 running Android 10.

**Comment by tobiasKaminsky** [comment]
Please see: https://github.com/nextcloud/android/issues/3623

**Comment by uamoti** [comment]
Funny fact: a previous installation on an OP1 running LineageOS 16 is still working despite the updates.
